docs: refactor local ZK stack docs (#176)
matter-labs/zksync-docs: content/10.zk-stack/20.running-a-zk-chain/20.using-a-local-zk-chain.md

a local reth node, or an Ethereum network (e.g., Sepolia).

### Base layer is the local reth node

If you chose to deploy on local [reth](https://ghcr.io/paradigmxyz/reth) node, you will have a set of addresses that have 100 ETH each.
You can find the list [here](https://github.com/matter-labs/local-setup/blob/main/rich-wallets.json)
and use these addresses to deposit into your ZK chain via the bridge.

### Base layer is an Ethereum network (e.g., Sepolia)

If you chose to deploy on an Ethereum network (e.g., Sepolia), you need to have an account on the base layer with ETH.
You can use the deployer, governor, or operator wallets setup during the the deployment process,
or any other one you have funds, to deposit into your ZK chain via the bridge.

Once you have the accounts with funds on the L1 base layer, you can do a deposit via the bridge to your ZK chain,
and any further interactions with your ZK chain.

## Using your ZK chain RPC

Your server contains both HTTPS as well as WebSocket (WS) services that are fully web3 compatible (and contain some extra ZK Stack functionalities).

> Learn more about it [here](/build/api-reference).

Reviewer: ```suggestion
Learn more on the [API reference page](/build/api-reference).
```

### Using zksync-cli

> ZKsync CLI allows you to easily interact and develop applications on your ZK chain.

Reviewer: ```suggestion
[ZKsync CLI](/build/zksync-cli) allows you to easily interact and develop applications on your ZK chain.
```

> When executing any command with zksync-cli, you can specify RPC urls for both L1 and L2.

Reviewer: ```suggestion
When executing any command with ZKsync CLI, you can specify RPC urls for both L1 and L2.
```

Your local server contains RPCs for both.
An example deposit command via the bridge would look like:

```bash
npx zksync-cli bridge deposit --rpc=http://localhost:3050 --l1-rpc=http://localhost:8545
```

> 

Reviewer: ```suggestion
You can save the chain configurations for your ZK chain with ZKsync CLI by [adding a new configuration](/build/zksync-cli/configuring-chains).
```

## Using the dApp Portal

The [dApp Portal](https://github.com/matter-labs/dapp-portal) module allows you to:

- Bridge & transfer tokens to your ZK chain.
- View balances.
- Add contacts for quick and easy access.

You can run the Portal module locally, and point it to your ZK chain configuration. It comes with scripts that help

> pulling the ZK chain configuration from your zksync-era repo and adapting to portal needs. Learn more

Reviewer: ```suggestion
pull the ZK chain configuration from your zksync-era repo and adapt it to portal needs. Learn more
```

> [here](https://github.com/matter-labs/dapp-portal). An example command would look like:

Reviewer: ```suggestion
on GitHub in the [matter-labs/dapp-portal project](https://github.com/matter-labs/dapp-portal). An example command would look like:
```

```bash
npm run hyperchain:configure ../zksync-era
npm run dev:node:hyperchain
```

You can now navigate to the displayed Portal URL (typically <http://localhost:3000>).

### Using Block Explorer

> A [free open source block explorer](https://github.com/matter-labs/block-explorer) is available for your ZK chain. Block explorer contains three components

Reviewer: ```suggestion
A [free open source block explorer](https://github.com/matter-labs/block-explorer) is available for your ZK chain. The block explorer contains three components
```

- [Worker](https://github.com/matter-labs/block-explorer/tree/main/packages/worker)
- [API](https://github.com/matter-labs/block-explorer/tree/main/packages/api)
- [App](https://github.com/matter-labs/block-explorer/tree/main/packages/app)

> Which you can run all together locally and connect to your ZK chain.

Reviewer: ```suggestion
You can run all components together locally and connect to your ZK chain.
```

Make sure you have your [zksync-era](https://github.com/matter-labs/zksync-era) repo set up locally and
the `zk_inception server` is running.

### Running block explorer locally

#### Install block explorer

Clone & install the block explorer repository:

```bash
cd /path/to/where/you/clone/repos
git clone https://github.com/matter-labs/block-explorer.git
cd block-explorer
npm install
```

#### Setting up env variables

> Next you need to set up all the necessary environment and configuration files with your ZK chain settings. You can use a script to set them up:

Reviewer: ```suggestion
Next you need to set up all the necessary environment and configuration files with your ZK chain settings. You can use the available npm script to set them up:
```

```bash
npm run hyperchain:configure
```

#### Run block explorer

Afterwards you can run the block explorer:

```bash
# if you are running block explorer for the first time
npm run db:create
```

```bash
npm run dev
```

#### Verify block explorer is up and running

> By default, you can access front-end `App` at <http://localhost:3010> in your browser. `API` should
> be available at <http://localhost:3020>, `Worker` at <http://localhost:3001> and `Data Fetcher` at <http://localhost:3040>.

Reviewer: ```suggestion
By default, you can access the front-end `App` at <http://localhost:3010> in your browser. The  `API` is
 available by default at <http://localhost:3020>, `Worker` at <http://localhost:3001> and `Data Fetcher` at <http://localhost:3040>.
```

---
title: Using a Local ZK chain
description:
---

## Funding accounts

The first step to start interacting with your ZK chain is to fund an account (or a few).
This means you need some funds on the base layer.

During the `zk_inception ecosystem create` configuration, you have a choice of what base layer to deploy the ZK chain onto:
a local reth node, or an Ethereum network (e.g., Sepolia).

### Base layer is the local reth node

If you choose to deploy on local [reth](https://ghcr.io/paradigmxyz/reth) node, you will have a set of addresses that have 100 ETH each.
You can find the list of rich wallets on GitHub in the [matter-labs/local-setup project](https://github.com/matter-labs/local-setup/blob/main/rich-wallets.json)
and use these addresses to deposit into your ZK chain via the bridge.

### Base layer is an Ethereum network (e.g., Sepolia)

If you choose to deploy on an Ethereum network (e.g., Sepolia), you need to have an account on the base layer with ETH.

> You can use the deployer, governor, or operator wallets setup during the the deployment process,

Reviewer 2: during **the the** deployment process
